src/tri.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2014-2024 bluss and ndarray developers.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use core::cmp::{max, min};
+
+use num_traits::Zero;
+
+use crate::{Array, ArrayBase, Axis, Data, Dimension, IntoDimension, RemoveAxis, Zip};
+
+impl<S, A, D> ArrayBase<S, D>
+where
+    S: Data<Elem = A>,
+    D: Dimension + RemoveAxis,
+    A: Clone + Zero,
+    D::Smaller: Copy,
```

Comment:
Is this required? I don't see any smaller dimension here.

Reply:
Ah ya that's left over from an intermediate implementation I was trying before this design, I think we can get rid of it.

Reply:
Ok coming back - this is required due to a trait bound on the `indexed` call.